src/pyspiffe/workloadapi/default_jwt_source.py

```diff
@@ -0,0 +1,150 @@
+
+    def get_jwt_svid(
+        self, audiences: Set[str], subject: Optional[SpiffeId] = None
+    ) -> JwtSvid:
+        """Returns an JWT-SVID from the source.
+
+        Args:
+            audiences: List of audiences for the JWT SVID.
+            subject: SPIFFE ID subject for the JWT.
+
+        Raises:
+            ArgumentError: In case audiences is empty.
+            JwtSourceError: In case an error is raised when connecting to WORKLOAD API.
+        """
+        if not audiences:
+            raise ArgumentError('Audience cannot be empty')
+        try:
+            jwt_svid = self._workload_api_client.fetch_jwt_svid(audiences, subject)
+            return jwt_svid
+        except Exception as err:
+            logging.error('JWT Source: error getting JWT SVID: {}.'.format(str(err)))
+            logging.error('JWT Source: closing due to invalid state.')
+            self.close()
+            raise JwtSourceError(
+                'JWT Source: error getting JWT SVID: {}'.format(str(err))
+            )
+
+    def get(self, trust_domain: TrustDomain) -> Optional[JwtBundle]:
+        """Returns the JWT bundle for the given trust domain.
+
+        Raises:
+            JwtSourceError: In case this JWT Source is closed.
+        """
+        with self._lock:
+            if self._closed:
+                raise JwtSourceError('Cannot get JWT Bundle: source is closed')
+            return self._jwt_bundle_set.get(trust_domain)
+
+    def close(self) -> None:
+        """Closes this JwtSource closing the underlying connection with the Workload API. Once the source is closed,
+        no methods can be called on it.
+
+        IMPORTANT: client code must call this method when the JwtSource is not needed anymore as the connection with Workload API will
+        only be closed when this method is invoked.
+        """
+        with self._lock:
+            # the cancel method throws a grpc exception, that can be discarded
+            try:
+                self._client_cancel_handler.cancel()
+            except Exception as err:
+                logging.exception(
+                    'Jwt Source: Exception canceling the Workload API client connection: {}'.format(
+                        str(err)
+                    )
+                )
+            self._initialized.set()
+            self._closed = True
+
+    def is_closed(self) -> bool:
+        """Returns True if the connection to Workload API is closed."""
+        with self._lock:
+            return self._closed
+
+    def _set_jwt_data(self, jwt_bundle_set: JwtBundleSet) -> None:
+        with self._lock:
+            self._jwt_bundle_set = jwt_bundle_set
+            self._initialized.set()
+
+    def _on_error(self, error: Exception) -> None:
+        self._log_error(error)
+        self.close()
+
+    def _log_error(self, err: Exception) -> None:
+        logging.error('JWT Source: Workload API client error: {}'.format(str(err)))
+        logging.error('JWT Source: closing.')
```

Comment:
Based on some changes introduced after you started this PR, you'll need to update the way you use the logger.
You need to define the logger for the module and then use it to log your messages.

```
_logger = logging.getLogger(__name__)
...
_logger.error('JWT Source: Workload API client error: {}'.format(str(err)))
```